sway-lsp/src/capabilities/code_actions/abi_impl.rs

```diff
@@ -0,0 +1,96 @@
+    let insertion_position = Position {
+        line: last_line as u32,
+        character: 0,
+    };
+    let text_edit = TextEdit {
+        range: Range {
+            start: insertion_position,
+            end: insertion_position,
+        },
+        new_text: get_contract_impl_string(abi_decl),
+    };
+    let mut text_edit_map = HashMap::new();
+    text_edit_map.insert(uri.clone(), vec![text_edit]);
+
+    CodeActionOrCommand::CodeAction(CodeAction {
+        title: String::from(CODE_ACTION_DESCRIPTION),
+        kind: Some(CodeActionKind::REFACTOR),
+        edit: Some(WorkspaceEdit {
+            changes: Some(text_edit_map),
+            ..Default::default()
+        }),
+        data: Some(Value::String(uri.to_string())),
+        ..Default::default()
+    })
+}
+
+fn get_function_signatures(abi_decl: TyAbiDeclaration) -> String {
+    abi_decl
+        .interface_surface
+        .iter()
+        .filter_map(|function_decl_id| {
+            declaration_engine::de_get_trait_fn(function_decl_id.clone(), &function_decl_id.span())
+                .ok()
+                .and_then(|function_decl| {
+                    let param_string: String = function_decl
+                        .parameters
+                        .iter()
+                        .map(|param| {
+                            format!("{}: {}", param.name.to_string(), param.type_id.to_string())
+                        })
+                        .collect::<Vec<String>>()
+                        .join(", ");
+                    let attribute_string = function_decl
+                        .attributes
+                        .iter()
+                        .map(|(_, attrs)| {
+                            attrs
+                                .iter()
+                                .map(|attr| format!("{}{}", TAB, attr.span.as_str().to_string()))
+                                .collect::<Vec<String>>()
+                                .join("\n")
+                        })
+                        .collect::<Vec<String>>()
+                        .join("\n");
+                    let attribute_prefix = match attribute_string.len() > 1 {
+                        true => "\n",
+                        false => "",
+                    };
+                    let signature = format!(
+                        "{}{}\n{}fn {}({}) -> {} {{}}",
+                        attribute_prefix,
+                        attribute_string,
+                        TAB,
+                        function_decl.name.clone().to_string(),
+                        param_string,
+                        function_decl.return_type.to_string()
+                    );
+                    Some(signature)
+                })
+        })
+        .collect::<Vec<String>>()
+        .join("\n")
+}
+
+fn get_contract_impl_string(abi_decl: TyAbiDeclaration) -> String {
+    let contract_name = abi_decl.name.to_string();
+    format!(
+        "\nimpl {} for Contract {{\n{}\n}}",
```

Comment:
Can we change this line to
```
"\nimpl {} for Contract {{{}\n}}",
```

Otherwise, there is an unnecessary empty line of code between the `impl` and the attributes of the first method

Reply:
Good catch!